tests/main.rs

```diff
@@ -148,12 +185,12 @@ fn overlapping(a: &MemArea, b: &MemArea) -> bool {
     // a_range is all elems in A
     let a_range = std::ops::Range {
         start: a.ptr,
-        end: a.ptr + a.len - 1,
+        end: a.ptr + a.len, // std::ops::Range is open from the right
     };
     // b_range is all elems in B
     let b_range = std::ops::Range {
         start: b.ptr,
-        end: b.ptr + b.len - 1,
+        end: b.ptr + b.len,
```

Comment:
Oh, and forgot to mention, I've also fixed a subtle bug in `fn overlapping`. It turns out `std::ops::Range { start, end }` is non-inclusive therefore it generates a set `[start, end)`, so there is no need to subtract 1 from `a.ptr + a.len` :-) Alternatively, we could use `std::ops::RangeInclusive` and leave the subtraction in place.

Reply:
Excellent thanks!